ducktape/cluster/cluster.py

```diff
@@ -16,14 +16,21 @@
 
 
 class ClusterSlot(object):
-    def __init__(self, parent, account, **kwargs):
-        self.parent = parent
+    def __init__(self, account, **kwargs):
         self.account = account
         for k, v in kwargs.items():
             setattr(self, k, v)
 
-    def free(self):
-        self.parent.free(self)
+    def __eq__(self, other):
+        if other is None:
```

Comment:
This implementation is fine, but given that its repeated it might be worth pointing out that you can just do `return other is not None and self.__dict__ == other.__dict__`.